src/FactoryGirl.js

```diff
@@ -6,72 +6,60 @@ import Factory from './Factory';
 import Sequence from './generators/Sequence';
 import Assoc from './generators/Assoc';
 import AssocAttrs from './generators/AssocAttrs';
-import Build from './generators/Build';
 import AssocMany from './generators/AssocMany';
 import AssocAttrsMany from './generators/AssocAttrsMany';
-import BuildMany from './generators/BuildMany';
 import ChanceGenerator from './generators/ChanceGenerator';
 import OneOf from './generators/OneOf';
-import attrGenerator from './generators/attrGenerator';
 import DefaultAdapter from './adapters/DefaultAdapter';
-// import _debug from 'debug';
 
-// const debug = _debug('FactoryGirl');
-
-class FactoryGirl {
+export default class FactoryGirl {
   factories = {};
   options = {};
   adapters = {};
   created = new Set();
 
   constructor(options = {}) {
-    this.assoc = attrGenerator(this, Assoc);
-    this.assocMany = attrGenerator(this, AssocMany);
-    this.assocBuild = attrGenerator(this, Build);
-    this.assocBuildMany = attrGenerator(this, BuildMany);
-    this.assocAttrs = attrGenerator(this, AssocAttrs);
-    this.assocAttrsMany = attrGenerator(this, AssocAttrsMany);
-    this.seq = this.sequence = attrGenerator(this, Sequence);
-    this.chance = attrGenerator(this, ChanceGenerator);
-    this.oneOf = attrGenerator(this, OneOf);
+    this.assoc = generatorThunk(this, Assoc);
+    this.assocMany = generatorThunk(this, AssocMany);
+    this.assocBuild = deprecate('assocBuild', 'assocAttrs');
+    this.assocBuildMany = deprecate('assocBuildMany', 'assocAttrsMany');
+    this.assocAttrs = generatorThunk(this, AssocAttrs);
+    this.assocAttrsMany = generatorThunk(this, AssocAttrsMany);
+    this.seq = this.sequence = generatorThunk(this, Sequence);
+    this.chance = generatorThunk(this, ChanceGenerator);
+    this.oneOf = generatorThunk(this, OneOf);
 
     this.defaultAdapter = new DefaultAdapter;
     this.options = options;
   }
 
   define(name, Model, initializer, options) {
     if (this.getFactory(name, false)) {
-      throw new Error(`factory ${name} already defined`);
+      throw new Error(`Factory ${name} already defined`);
     }
-
     this.factories[name] = new Factory(Model, initializer, options);
   }
 
-  attrs(name, attrs, buildOptions) {
+  async attrs(name, attrs, buildOptions) {
```

Comment:
Just a note: I left out `async` because the returned value was a promise anyway. No problem to keep it though.